db/model/Gdoc/Gdoc.ts

```diff
@@ -183,6 +193,58 @@ export class Gdoc extends BaseEntity implements OwidArticleType {
         this.linkedDocuments = keyBy(linkedDocuments, "id")
     }
 
+    async loadLinkedCharts(
+        publishedExplorersBySlug: Record<string, any>
+    ): Promise<void> {
+        const slugToIdMap = await Chart.mapSlugsToIds()
+        const uniqueSlugsByLinkType = this.links
+            .filter(({ linkType }) =>
+                ["grapher", "explorer"].includes(linkType)
+            )
+            .reduce(
+                (acc, { target, linkType }) => {
+                    if (!acc[linkType].includes(target)) {
+                        acc[linkType].push(target)
+                    }
+                    return acc
+                },
+                { grapher: [], explorer: [] } as Record<string, string[]>
+            )
```

Comment:
As far as I can see, the only difference to `keyBy` is that you keep the first occurrence, rather than the last one as `keyBy` does. Is that absolutely necessary, or would `keyBy` work just fine here?

Reply:
I'm not sure what you mean. Maybe there's been a misunderstanding of this code?

I have:
```js
[
    {
        linkType: "grapher",
        target: "life-expectancy",
    },
    {
        linkType: "grapher",
        target: "gdp",
    },
    {
        linkType: "explorer",
        target: "food-footprints",
    },
    {
        linkType: "grapher",
        target: "literacy-rates",
    },
]
```

I want

```js
{
    grapher: ["life-expectancy", "gdp", "literacy-rates"],
    explorer: ["food-footprints"],
}
```

`groupBy` would kind of work but then I'd also have to map and filter each group so this reduce seemed simpler.

Reply:
Actually, I should just use a set here. That would be way simpler.

Reply:
Ah, right, I see now. But 👍🏻 for sets.